src/tests/ftest/dfuse/pil4dfs_dcache.py

```diff
@@ -0,0 +1,329 @@
+            {
+                "test_name": "test_open_close",
+                "test_id": 4,
+                "dcache_add": 0,
+                "dcache_del": 0,
+                "dcache_hit": 0,
+                "dcache_miss": 0,
+                "no_dcache_new": 2,
+                "no_dcache_del": 2,
+                "op_name": None
+            },
+            {
+                "test_name": "test_dup",
+                "test_id": 5,
+                "dcache_add": 0,
+                "dcache_del": 0,
+                "dcache_hit": 0,
+                "dcache_miss": 0,
+                "no_dcache_new": 2,
+                "no_dcache_del": 2,
+                "op_name": None
+            }
+        ],
+    }
+
+    __dcache_re__ = {
+        "dcache_add": re.compile(r'^.+ il +DBUG .+ dcache_add\(\) .+$'),
+        "dcache_del": re.compile(r'^.+ il +DBUG .+ dcache_rec_free\(\) .+$'),
+        "dcache_hit": re.compile(r'^.+ il +DBUG .+ dcache hit:.+$'),
+        "dcache_miss": re.compile(r'^.+ il +DBUG .+ dcache miss:.+$'),
+        "no_dcache_new": re.compile(r'^.+ il +DBUG .+ dcache_find_insert_dact\(\).+$'),
+        "no_dcache_del": re.compile(r'^.+ il +DBUG .+ drec_del_at_dact\(\) .+$')
+    }
+
+    __start_test_re__ = re.compile(r'^-- START of test_.+ --$')
+    __end_test_re__ = re.compile(r'^-- END of test_.+ --$')
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+
+        # Start the servers and agents
+        super().setUp()
+
+    def mount_dfuse(self):
+        """Mount a DFuse mount point."""
+        self.log.info("Creating DAOS pool")
+        self.add_pool()
+
+        self.log.info("Creating DAOS container")
+        self.add_container(self.pool)
+
+        self.log.info("Mounting DFuse mount point")
+        self.start_dfuse(self.hostlist_clients, self.pool, self.container)
+
+    def update_cmd_env(self, env, mnt, timeout=None):
+        """Update the Pil4dfsDcacheCmd command environment.
+
+        Args:
+            env (dict): dictionary of the command environment variables
+            mnt (str): path of the Dfuse mount point
+            timeout (int, optional): dir-cache timeout in seconds.
+                Default is None.
+        """
+        lib_dir = os.path.join(self.prefix, 'lib64', 'libpil4dfs.so')
+        env['LD_PRELOAD'] = lib_dir
+        env['D_DFUSE_MNT'] = mnt
+        env['D_LOG_MASK'] = 'DEBUG'
+        env['DD_SUBSYS'] = 'il'
+        env['DD_MASK'] = 'trace'
+        env['D_IL_REPORT'] = '1'
+
+        if timeout is not None:
+            env["D_IL_DCACHE_REC_TIMEOUT"] = timeout
+
+    def check_result(self, test_case, lines):
+        """Check consistency of the test output.
+
+        Args:
+            test_case (dict): dictionary of the expected output result
+            lines (list): list of the test command stdout lines
+        """
+        dcache_count = {key: 0 for key in Pil4dfsDcache.__dcache_re__}
```

Comment:
Perhaps I'm mis-reading this but can't this (and lots of other instances) be accessed via self rather than the class name?
```suggestion
        dcache_count = {key: 0 for key in self.__dcache_re__}
```

Reply:
Nope, it can not be accessed through self, as they are class attributes: they are common to all class instances.